File: src/main/java/me/miki/shindo/management/settings/impl/combo/Option.java
package me.miki.shindo.management.settings.impl.combo;

import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.settings.Setting;

public class Option {

    private final TranslateText nameTranslate;
    private final String fallbackName;
    private final String nameKey;

    public Option(TranslateText nameTranslate) {
        this.nameTranslate = nameTranslate;
        this.fallbackName = nameTranslate.getText();
        this.nameKey = nameTranslate.getKey();
    }

    public Option(String name) {
        this.nameTranslate = null;
        this.fallbackName = name;
        this.nameKey = buildKey(name);
    }

    public String getName() {
        return nameTranslate != null ? nameTranslate.getText() : fallbackName;
    }

    public String getNameKey() {
        return nameKey;
    }

    public TranslateText getTranslate() {
        return nameTranslate;
    }

    private static String buildKey(String raw) {
        return raw == null ? "" : Setting.normalizeKey(raw);
    }
}

File: src/main/kotlin/me/miki/shindo/management/settings/impl/combo/Option.kt
package me.miki.shindo.management.settings.impl.combo

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.settings.Setting

class Option {

    private val nameTranslate: TranslateText?
    private val fallbackName: String
    val nameKey: String

    constructor(nameTranslate: TranslateText) {
        this.nameTranslate = nameTranslate
        this.fallbackName = nameTranslate.text
        this.nameKey = nameTranslate.key
    }

    constructor(name: String) {
        this.nameTranslate = null
        this.fallbackName = name
        this.nameKey = buildKey(name)
    }

    val name: String
        get() = nameTranslate?.text ?: fallbackName

    fun getTranslate(): TranslateText? {
        return nameTranslate
    }

    private fun buildKey(raw: String?): String {
        return if (raw == null) "" else Setting.normalizeKey(raw)
    }
}
